File: tests/snapshots/split_join.rs
// Split, Join, Space — VB's string-list builtins.

fn vbr_main() -> Result<(), String> {
    let csv: String = "one,two,three".to_string();
    let parts: Vec<String> = csv.split(",").map(|__p| __p.to_string()).collect::<Vec<_>>();
    println!("{}", parts.join(" / "));
    // Default delimiter is a single space, both ways.
    println!("{}", "a b c".split(' ').map(|__p| __p.to_string()).collect::<Vec<_>>().join(" "));
    println!("[{}]", " ".repeat(((3) as i64).max(0) as usize));
    Ok(())
}

fn main() {
    if let Err(error) = vbr_main() {
        eprintln!("Error: {error}");
        std::process::exit(1);
    }
}

File: tests/snapshots/split_join.py
# Split, Join, Space — VB's string-list builtins.

def _vb(x):
    if isinstance(x, bool):
        return "true" if x else "false"
    if isinstance(x, float) and x.is_integer():
        return str(int(x))
    return str(x)

def main():
    csv: str = 'one,two,three'
    parts: list[str] = (csv).split(',')
    print(_vb((' / ').join(parts)))
    # Default delimiter is a single space, both ways.
    print(_vb(' '.join(('a b c').split(' '))))
    print(f"[{_vb((' ' * max(int(3), 0)))}]")


if __name__ == "__main__":
    main()
